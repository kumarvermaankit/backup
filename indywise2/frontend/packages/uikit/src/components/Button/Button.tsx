import * as React from 'react';
import styled from 'styled-components';

import { Icon, IconTypes } from '../Icons';

export interface IButtonProps {
  /**
   * Text to be displayed on the button. You can also use `props.children`.
   *
   * @example `My Button`
   */
  text?: string;

  /**
   * We have three color designs, `primary`, `secondary` and `tertiary`.
   *
   * @default `primary`
   */
  color?: 'primary' | 'secondary' | 'tertiary';

  /**
   * Callback function to be triggered on `onClick` event on the button.
   */
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  onClick?: (e?: any) => any;

  /**
   * If set to `true`, the button will not execute `onClick` event to deny spamming
   * of the button and will also display a loading animation.
   */
  isLoading?: boolean;

  /**
   * If set to `true`, the button will be disabled.
   * @default false
   */
  isDisabled?: boolean;

  /**
   * Custom CSS style. Applies to the `ButtonText` element.
   */
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  fontStyle?: any;

  /**
   * Icon to be rendered in the button. It can a string which should be a valid
   * string based on `IconTypes` or it can be a React Component.
   */
  icon?: IconTypes | React.ReactElement | string;

  /**
   * Defines the color of the Icon rendered on the button
   */
  iconColor?: string;

  /**
   * Icon to be rendered left or right side of the text.
   *
   * @default left
   */
  iconAlign?: 'left' | 'right';

  /**
   * Icon size.
   *
   * @default `100%``
   */
  iconSize?: string;

  /**
   * Roate the icon of the button.
   */
  iconRotate?: number;

  /**
   * Custom CSS style. Applies to the `ButtonBody` element.
   */
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  style?: any;
}

export const Button: React.FC<IButtonProps> = (props) => {
  const color = props.color || 'primary';
  const style = props.style || {};
  const fontStyle = props.fontStyle || {};
  // Text for the button can be passed as a prop or a children.
  const text = props.text ? props.text : props.children;

  if (props.isDisabled) {
    return (
      <DisabledButton style={style} color={color}>
        <ButtonText style={fontStyle} color={color}>
          {text}
        </ButtonText>
      </DisabledButton>
    );
  }

  let onClick;
  // The callback function which is triggered on `onClick` event will only be
  // executed if `isLoading` and `isDisabled` are both `false`.
  if (!props.isLoading && !props.isDisabled) {
    onClick = props.onClick;
  }

  let icon;
  const iconAlign = props.iconAlign || 'left';
  const iconSize = props.iconSize || '24px';
  const iconColor = props.iconColor;
  const iconRotate = props.iconRotate;
  if (typeof props.icon === 'string' && props.icon) {
    icon = (
      <Icon
        icon={props.icon as IconTypes}
        size={iconSize}
        color={iconColor}
        rotate={iconRotate}
      />
    );
  } else if (typeof props.icon !== 'string' && props.icon) {
    icon = props.icon;
  }

  let contents;
  if (icon && iconAlign === 'right') {
    contents = (
      <ContentWrapper>
        <ButtonText style={fontStyle} color={color}>
          {text}
        </ButtonText>
        <i style={{ marginLeft: '0.5em', height: `${iconSize}` }}>{icon}</i>
      </ContentWrapper>
    );
  } else {
    contents = (
      <ContentWrapper>
        <i style={{ marginRight: '0.5em', height: `${iconSize}` }}>{icon}</i>
        <ButtonText style={fontStyle} color={color}>
          {text}
        </ButtonText>
      </ContentWrapper>
    );
  }

  if (icon === undefined) {
    contents = (
      <ContentWrapper>
        <ButtonText style={fontStyle} color={color}>
          {text}
        </ButtonText>
      </ContentWrapper>
    );
  }

  return (
    <ButtonBody style={style} color={color} onClick={onClick}>
      {contents}
    </ButtonBody>
  );
};

const ContentWrapper = styled.span`
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0 auto;
`;

const ButtonBody = styled.button<{
  color: 'primary' | 'secondary' | 'tertiary';
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  style: any;
}>`
  background-color: ${(props) =>
    props.color === 'primary'
      ? '#F2A922'
      : props.color === 'secondary'
      ? '#FFFFFF'
      : '#0C3559'};
  border: 0;
  border-radius: 8px;
  padding: 8px 16px;
  filter: ${(props) =>
    props.color === 'primary'
      ? 'drop-shadow(0px 8px 16px rgba(242, 169, 34, 0.32))'
      : props.color === 'secondary'
      ? 'drop-shadow(0px 8px 16px rgba(17, 17, 17, 0.16))'
      : 'box-shadow: 0px 8px 16px rgba(17, 17, 17, 0.16)'};

  &:hover {
    cursor: pointer;
    border: ${(props) =>
      props.color === 'primary'
        ? 'none'
        : props.color === 'secondary'
        ? '1px solid #F2A922'
        : 'none'};
    background-color: ${(props) =>
      props.color === 'primary'
        ? '#F8BD4F'
        : props.color === 'secondary'
        ? '#FFFFFF'
        : '#19588F'};
    filter: ${(props) =>
      props.color === 'primary'
        ? 'drop-shadow(0px 16px 32px rgba(242, 169, 34, 0.24))'
        : props.color === 'secondary'
        ? 'drop-shadow(0px 16px 32px rgba(17, 17, 17, 0.08))'
        : 'box-shadow: 0px 16px 32px rgba(17, 17, 17, 0.08)'};
  }

  &:focus {
    box-shadow: 0 0 0 2px
        ${(props) =>
          props.color === 'primary'
            ? '#F8BD4F'
            : props.color === 'secondary'
            ? '#E9E9E9'
            : '#90C0EA'},
      0 0 3px 5px
        ${(props) =>
          props.color === 'primary'
            ? '#CB870A'
            : props.color === 'secondary'
            ? '#727272'
            : '#90C0EA'};
    outline: 2px dotted transparent;
    outline-offset: 2px;
  }

  &:active {
    filter: ${(props) =>
      props.color === 'primary'
        ? 'drop-shadow(0px 4px 8px rgba(242, 169, 34, 0.4))'
        : props.color === 'secondary'
        ? 'drop-shadow(0px 4px 8px rgba(17, 17, 17, 0.32))'
        : 'box-shadow: 0px 4px 8px rgba(17, 17, 17, 0.32)'};
  }
`;

// A seperate button because we dont care about `active` state, also to keep
// `ButtonBody` a bit clean by not adding `props.isDisabled` check everywhere.
const DisabledButton = styled.button<{ color: string }>`
  background-color: ${(props) =>
    props.color === 'primary'
      ? '#F8BD4F'
      : props.color === 'secondary'
      ? '#E9E9E9'
      : '#EAF3FA'};
  border: 0;
  border-radius: 8px;
  padding: 8px 16px;

  p {
    color: #727272;
  }

  &:hover {
    cursor: not-allowed;
  }

  &:focus {
    outline: none;
  }
`;

const ButtonText = styled.p<{ color: 'primary' | 'secondary' | 'tertiary' }>`
  margin: auto 0;
  font-family: 'Roboto', 'sans-serif';
  font-size: 16px;
  line-height: 28px;
  color: ${(props) =>
    props.color === 'primary'
      ? '#042039'
      : props.color === 'secondary'
      ? '#0C3559'
      : '#FFFFFF'};
`;
