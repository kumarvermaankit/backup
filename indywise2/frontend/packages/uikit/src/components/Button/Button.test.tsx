import * as React from 'react';
import { create } from 'react-test-renderer';

import { Button } from './Button';

describe('Button', () => {
  it('Disable Primary', () => {
    const component = create(
      <Button text="Button" color="primary" isDisabled={true}>
        Button
      </Button>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Disable Secondary', () => {
    const component = create(
      <Button text="Button" color="secondary" isDisabled={true}>
        Button
      </Button>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Disable Tertiary', () => {
    const component = create(
      <Button text="Button" color="tertiary" isDisabled={true}>
        Button
      </Button>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Primary', () => {
    const component = create(
      <Button text="Button" color="primary">
        Button
      </Button>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Secondary', () => {
    const component = create(
      <Button text="Button" color="secondary">
        Button
      </Button>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Tertiary', () => {
    const component = create(
      <Button text="Button" color="tertiary">
        Button
      </Button>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Primary with Icon props left align', () => {
    const component = create(
      <Button
        text="Button"
        color="primary"
        icon="download"
        iconAlign="left"
        iconColor="#0C3559"
        iconSize="24px"
      >
        Button
      </Button>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Primary with Icon props right align', () => {
    const component = create(
      <Button
        text="Button"
        color="primary"
        icon="download"
        iconAlign="right"
        iconColor="#0C3559"
        iconSize="24px"
      >
        Button
      </Button>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Secondary with Icon props left align', () => {
    const component = create(
      <Button
        text="Button"
        color="secondary"
        icon="download"
        iconAlign="left"
        iconColor="#262626"
        iconSize="24px"
      >
        Button
      </Button>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Secondary with Icon props right align', () => {
    const component = create(
      <Button
        text="Button"
        color="secondary"
        icon="download"
        iconAlign="right"
        iconColor="#262626"
        iconSize="24px"
      >
        Button
      </Button>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Tertiary with Icon props left align', () => {
    const component = create(
      <Button
        text="Button"
        color="tertiary"
        icon="download"
        iconAlign="left"
        iconColor="#FFFFFF"
        iconSize="24px"
      >
        Button
      </Button>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Tertiary with Icon props right align', () => {
    const component = create(
      <Button
        text="Button"
        color="tertiary"
        icon="download"
        iconAlign="right"
        iconColor="#FFFFFF"
        iconSize="24px"
      >
        Button
      </Button>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Styled Button', () => {
    const component = create(
      <Button
        style={{
          backgroundColor: '#222',
          color: '#fff',
          border: '1px solid #F2A922'
        }}
      >
        Button
      </Button>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Font styled', () => {
    const component = create(
      <Button
        fontStyle={{
          fontFamily: 'Robot',
          fontSize: '20px',
          lineHeight: '24px',
          color: '#333'
        }}
      >
        Button
      </Button>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Loading Button', () => {
    const component = create(<Button isLoading={true}>Button</Button>).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Click fun', () => {
    const component = create(
      <Button onClick={() => console.log('Button Clicked!')}>Button</Button>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Icon rotation', () => {
    const component = create(
      <Button icon="arrow" iconRotate={-90} iconColor="#fff">
        Button
      </Button>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('No Type and Text', () => {
    const component = create(
      <Button
        icon="download"
        iconAlign="right"
        iconColor="#262626"
        iconSize="24px"
      >
        Button
      </Button>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('only icon', () => {
    const component = create(<Button icon="download">Button</Button>).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('React Element icon', () => {
    const component = create(
      <Button icon={<div>Hello</div>}>Button</Button>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
