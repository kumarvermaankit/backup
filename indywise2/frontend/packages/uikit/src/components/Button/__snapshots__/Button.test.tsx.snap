// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Click fun 1`] = `
<button
  className=""
  color="primary"
  onClick={[Function]}
  style={Object {}}
>
  <span
    className=""
  >
    <p
      className=""
      color="primary"
      style={Object {}}
    >
      Button
    </p>
  </span>
</button>
`;

exports[`Button Disable Primary 1`] = `
<button
  className=""
  color="primary"
  style={Object {}}
>
  <p
    className=""
    color="primary"
    style={Object {}}
  >
    Button
  </p>
</button>
`;

exports[`Button Disable Secondary 1`] = `
<button
  className=""
  color="secondary"
  style={Object {}}
>
  <p
    className=""
    color="secondary"
    style={Object {}}
  >
    Button
  </p>
</button>
`;

exports[`Button Disable Tertiary 1`] = `
<button
  className=""
  color="tertiary"
  style={Object {}}
>
  <p
    className=""
    color="tertiary"
    style={Object {}}
  >
    Button
  </p>
</button>
`;

exports[`Button Font styled 1`] = `
<button
  className=""
  color="primary"
  style={Object {}}
>
  <span
    className=""
  >
    <p
      className=""
      color="primary"
      style={
        Object {
          "color": "#333",
          "fontFamily": "Robot",
          "fontSize": "20px",
          "lineHeight": "24px",
        }
      }
    >
      Button
    </p>
  </span>
</button>
`;

exports[`Button Icon rotation 1`] = `
<button
  className=""
  color="primary"
  style={Object {}}
>
  <span
    className=""
  >
    <i
      style={
        Object {
          "height": "24px",
          "marginRight": "0.5em",
        }
      }
    >
      <span
        className=""
        color="#fff"
        rotate={-90}
        size="24px"
        style={Object {}}
      >
        <svg
          fill="none"
          height="7"
          viewBox="0 0 14 7"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M12.2312 0.381507C12.4387 0.199371 12.767 0.208218 12.9645 0.401268C13.1619 0.594319 13.1537 0.898467 12.9462 1.0806L7.37582 5.96865C7.24646 6.08216 7.0702 6.12149 6.90795 6.08829C6.80683 6.07429 6.70927 6.03257 6.62952 5.96273L1.0539 1.07986C0.84616 0.897934 0.837974 0.594114 1.03562 0.401261C1.23326 0.208409 1.56189 0.199555 1.76963 0.381484L7.00565 4.96694L12.2312 0.381507Z"
            fill="#0C3559"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </i>
    <p
      className=""
      color="primary"
      style={Object {}}
    >
      Button
    </p>
  </span>
</button>
`;

exports[`Button Loading Button 1`] = `
<button
  className=""
  color="primary"
  style={Object {}}
>
  <span
    className=""
  >
    <p
      className=""
      color="primary"
      style={Object {}}
    >
      Button
    </p>
  </span>
</button>
`;

exports[`Button No Type and Text 1`] = `
<button
  className=""
  color="primary"
  style={Object {}}
>
  <span
    className=""
  >
    <p
      className=""
      color="primary"
      style={Object {}}
    >
      Button
    </p>
    <i
      style={
        Object {
          "height": "24px",
          "marginLeft": "0.5em",
        }
      }
    >
      <span
        className=""
        color="#262626"
        size="24px"
        style={Object {}}
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.7071 17.7071C12.3166 18.0976 11.6834 18.0976 11.2929 17.7071L4.92893 11.3431C4.53841 10.9526 4.53841 10.3195 4.92893 9.92893C5.31946 9.53841 5.95262 9.53841 6.34315 9.92893L12 15.5858L17.6569 9.92893C18.0474 9.53841 18.6805 9.53841 19.0711 9.92893C19.4616 10.3195 19.4616 10.9526 19.0711 11.3431L12.7071 17.7071ZM4 20C3.44772 20 3 19.5523 3 19C3 18.4477 3.44772 18 4 18L4 20ZM20 18C20.5523 18 21 18.4477 21 19C21 19.5523 20.5523 20 20 20V18ZM13 3V17H11V3H13ZM4 18L20 18V20L4 20L4 18Z"
            fill="#262626"
          />
        </svg>
      </span>
    </i>
  </span>
</button>
`;

exports[`Button Primary 1`] = `
<button
  className=""
  color="primary"
  style={Object {}}
>
  <span
    className=""
  >
    <p
      className=""
      color="primary"
      style={Object {}}
    >
      Button
    </p>
  </span>
</button>
`;

exports[`Button Primary with Icon props left align 1`] = `
<button
  className=""
  color="primary"
  style={Object {}}
>
  <span
    className=""
  >
    <i
      style={
        Object {
          "height": "24px",
          "marginRight": "0.5em",
        }
      }
    >
      <span
        className=""
        color="#0C3559"
        size="24px"
        style={Object {}}
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.7071 17.7071C12.3166 18.0976 11.6834 18.0976 11.2929 17.7071L4.92893 11.3431C4.53841 10.9526 4.53841 10.3195 4.92893 9.92893C5.31946 9.53841 5.95262 9.53841 6.34315 9.92893L12 15.5858L17.6569 9.92893C18.0474 9.53841 18.6805 9.53841 19.0711 9.92893C19.4616 10.3195 19.4616 10.9526 19.0711 11.3431L12.7071 17.7071ZM4 20C3.44772 20 3 19.5523 3 19C3 18.4477 3.44772 18 4 18L4 20ZM20 18C20.5523 18 21 18.4477 21 19C21 19.5523 20.5523 20 20 20V18ZM13 3V17H11V3H13ZM4 18L20 18V20L4 20L4 18Z"
            fill="#262626"
          />
        </svg>
      </span>
    </i>
    <p
      className=""
      color="primary"
      style={Object {}}
    >
      Button
    </p>
  </span>
</button>
`;

exports[`Button Primary with Icon props right align 1`] = `
<button
  className=""
  color="primary"
  style={Object {}}
>
  <span
    className=""
  >
    <p
      className=""
      color="primary"
      style={Object {}}
    >
      Button
    </p>
    <i
      style={
        Object {
          "height": "24px",
          "marginLeft": "0.5em",
        }
      }
    >
      <span
        className=""
        color="#0C3559"
        size="24px"
        style={Object {}}
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.7071 17.7071C12.3166 18.0976 11.6834 18.0976 11.2929 17.7071L4.92893 11.3431C4.53841 10.9526 4.53841 10.3195 4.92893 9.92893C5.31946 9.53841 5.95262 9.53841 6.34315 9.92893L12 15.5858L17.6569 9.92893C18.0474 9.53841 18.6805 9.53841 19.0711 9.92893C19.4616 10.3195 19.4616 10.9526 19.0711 11.3431L12.7071 17.7071ZM4 20C3.44772 20 3 19.5523 3 19C3 18.4477 3.44772 18 4 18L4 20ZM20 18C20.5523 18 21 18.4477 21 19C21 19.5523 20.5523 20 20 20V18ZM13 3V17H11V3H13ZM4 18L20 18V20L4 20L4 18Z"
            fill="#262626"
          />
        </svg>
      </span>
    </i>
  </span>
</button>
`;

exports[`Button React Element icon 1`] = `
<button
  className=""
  color="primary"
  style={Object {}}
>
  <span
    className=""
  >
    <i
      style={
        Object {
          "height": "24px",
          "marginRight": "0.5em",
        }
      }
    >
      <div>
        Hello
      </div>
    </i>
    <p
      className=""
      color="primary"
      style={Object {}}
    >
      Button
    </p>
  </span>
</button>
`;

exports[`Button Secondary 1`] = `
<button
  className=""
  color="secondary"
  style={Object {}}
>
  <span
    className=""
  >
    <p
      className=""
      color="secondary"
      style={Object {}}
    >
      Button
    </p>
  </span>
</button>
`;

exports[`Button Secondary with Icon props left align 1`] = `
<button
  className=""
  color="secondary"
  style={Object {}}
>
  <span
    className=""
  >
    <i
      style={
        Object {
          "height": "24px",
          "marginRight": "0.5em",
        }
      }
    >
      <span
        className=""
        color="#262626"
        size="24px"
        style={Object {}}
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.7071 17.7071C12.3166 18.0976 11.6834 18.0976 11.2929 17.7071L4.92893 11.3431C4.53841 10.9526 4.53841 10.3195 4.92893 9.92893C5.31946 9.53841 5.95262 9.53841 6.34315 9.92893L12 15.5858L17.6569 9.92893C18.0474 9.53841 18.6805 9.53841 19.0711 9.92893C19.4616 10.3195 19.4616 10.9526 19.0711 11.3431L12.7071 17.7071ZM4 20C3.44772 20 3 19.5523 3 19C3 18.4477 3.44772 18 4 18L4 20ZM20 18C20.5523 18 21 18.4477 21 19C21 19.5523 20.5523 20 20 20V18ZM13 3V17H11V3H13ZM4 18L20 18V20L4 20L4 18Z"
            fill="#262626"
          />
        </svg>
      </span>
    </i>
    <p
      className=""
      color="secondary"
      style={Object {}}
    >
      Button
    </p>
  </span>
</button>
`;

exports[`Button Secondary with Icon props right align 1`] = `
<button
  className=""
  color="secondary"
  style={Object {}}
>
  <span
    className=""
  >
    <p
      className=""
      color="secondary"
      style={Object {}}
    >
      Button
    </p>
    <i
      style={
        Object {
          "height": "24px",
          "marginLeft": "0.5em",
        }
      }
    >
      <span
        className=""
        color="#262626"
        size="24px"
        style={Object {}}
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.7071 17.7071C12.3166 18.0976 11.6834 18.0976 11.2929 17.7071L4.92893 11.3431C4.53841 10.9526 4.53841 10.3195 4.92893 9.92893C5.31946 9.53841 5.95262 9.53841 6.34315 9.92893L12 15.5858L17.6569 9.92893C18.0474 9.53841 18.6805 9.53841 19.0711 9.92893C19.4616 10.3195 19.4616 10.9526 19.0711 11.3431L12.7071 17.7071ZM4 20C3.44772 20 3 19.5523 3 19C3 18.4477 3.44772 18 4 18L4 20ZM20 18C20.5523 18 21 18.4477 21 19C21 19.5523 20.5523 20 20 20V18ZM13 3V17H11V3H13ZM4 18L20 18V20L4 20L4 18Z"
            fill="#262626"
          />
        </svg>
      </span>
    </i>
  </span>
</button>
`;

exports[`Button Styled Button 1`] = `
<button
  className=""
  color="primary"
  style={
    Object {
      "backgroundColor": "#222",
      "border": "1px solid #F2A922",
      "color": "#fff",
    }
  }
>
  <span
    className=""
  >
    <p
      className=""
      color="primary"
      style={Object {}}
    >
      Button
    </p>
  </span>
</button>
`;

exports[`Button Tertiary 1`] = `
<button
  className=""
  color="tertiary"
  style={Object {}}
>
  <span
    className=""
  >
    <p
      className=""
      color="tertiary"
      style={Object {}}
    >
      Button
    </p>
  </span>
</button>
`;

exports[`Button Tertiary with Icon props left align 1`] = `
<button
  className=""
  color="tertiary"
  style={Object {}}
>
  <span
    className=""
  >
    <i
      style={
        Object {
          "height": "24px",
          "marginRight": "0.5em",
        }
      }
    >
      <span
        className=""
        color="#FFFFFF"
        size="24px"
        style={Object {}}
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.7071 17.7071C12.3166 18.0976 11.6834 18.0976 11.2929 17.7071L4.92893 11.3431C4.53841 10.9526 4.53841 10.3195 4.92893 9.92893C5.31946 9.53841 5.95262 9.53841 6.34315 9.92893L12 15.5858L17.6569 9.92893C18.0474 9.53841 18.6805 9.53841 19.0711 9.92893C19.4616 10.3195 19.4616 10.9526 19.0711 11.3431L12.7071 17.7071ZM4 20C3.44772 20 3 19.5523 3 19C3 18.4477 3.44772 18 4 18L4 20ZM20 18C20.5523 18 21 18.4477 21 19C21 19.5523 20.5523 20 20 20V18ZM13 3V17H11V3H13ZM4 18L20 18V20L4 20L4 18Z"
            fill="#262626"
          />
        </svg>
      </span>
    </i>
    <p
      className=""
      color="tertiary"
      style={Object {}}
    >
      Button
    </p>
  </span>
</button>
`;

exports[`Button Tertiary with Icon props right align 1`] = `
<button
  className=""
  color="tertiary"
  style={Object {}}
>
  <span
    className=""
  >
    <p
      className=""
      color="tertiary"
      style={Object {}}
    >
      Button
    </p>
    <i
      style={
        Object {
          "height": "24px",
          "marginLeft": "0.5em",
        }
      }
    >
      <span
        className=""
        color="#FFFFFF"
        size="24px"
        style={Object {}}
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.7071 17.7071C12.3166 18.0976 11.6834 18.0976 11.2929 17.7071L4.92893 11.3431C4.53841 10.9526 4.53841 10.3195 4.92893 9.92893C5.31946 9.53841 5.95262 9.53841 6.34315 9.92893L12 15.5858L17.6569 9.92893C18.0474 9.53841 18.6805 9.53841 19.0711 9.92893C19.4616 10.3195 19.4616 10.9526 19.0711 11.3431L12.7071 17.7071ZM4 20C3.44772 20 3 19.5523 3 19C3 18.4477 3.44772 18 4 18L4 20ZM20 18C20.5523 18 21 18.4477 21 19C21 19.5523 20.5523 20 20 20V18ZM13 3V17H11V3H13ZM4 18L20 18V20L4 20L4 18Z"
            fill="#262626"
          />
        </svg>
      </span>
    </i>
  </span>
</button>
`;

exports[`Button only icon 1`] = `
<button
  className=""
  color="primary"
  style={Object {}}
>
  <span
    className=""
  >
    <i
      style={
        Object {
          "height": "24px",
          "marginRight": "0.5em",
        }
      }
    >
      <span
        className=""
        size="24px"
        style={Object {}}
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.7071 17.7071C12.3166 18.0976 11.6834 18.0976 11.2929 17.7071L4.92893 11.3431C4.53841 10.9526 4.53841 10.3195 4.92893 9.92893C5.31946 9.53841 5.95262 9.53841 6.34315 9.92893L12 15.5858L17.6569 9.92893C18.0474 9.53841 18.6805 9.53841 19.0711 9.92893C19.4616 10.3195 19.4616 10.9526 19.0711 11.3431L12.7071 17.7071ZM4 20C3.44772 20 3 19.5523 3 19C3 18.4477 3.44772 18 4 18L4 20ZM20 18C20.5523 18 21 18.4477 21 19C21 19.5523 20.5523 20 20 20V18ZM13 3V17H11V3H13ZM4 18L20 18V20L4 20L4 18Z"
            fill="#262626"
          />
        </svg>
      </span>
    </i>
    <p
      className=""
      color="primary"
      style={Object {}}
    >
      Button
    </p>
  </span>
</button>
`;
